db-service/lib/SQLService.js

```diff
@@ -192,7 +192,27 @@ class SQLService extends DatabaseService {
     // REVISIT: It's not yet 100 % clear under which circumstances we can rely on db constraints
     return (super.onDELETE = /* cds.env.features.assert_integrity === 'db' ? this.onSIMPLE : */ deep_delete)
     async function deep_delete(/** @type {Request} */ req) {
-      let { compositions } = req.target
+      const transitions = getTransition(req.query.target, this)
+      if (transitions.target !== transitions.queryTarget) {
+        const matchedKeys = Object.keys(transitions.queryTarget.keys)
+          .filter(key => transitions.mapping.has(key))
+          .map(k => ({ ref: [k] }))
+        const query = DELETE.from({
+          ref: [
+            {
+              id: transitions.target.name,
+              where: [
+                { list: matchedKeys.map(k => transitions.mapping.get(k.ref[0])) },
```

Comment:
the map result still has to be `{ref:[...]}`.

Reply:
it returned `{ref:[...]}` correctly but in the case of `IsActiveEntity` we got an error because it could not be found in the select statement

Reply:
Does this line produce `[{ref:[<col>]},...]` or does this produce `[<col>,...]` ?

Reply:
@BobdenOs 
`matchedKeys.map(k => transitions.mapping.get(k.ref[0])) ` returns `[ { ref: [ 'ID' ] } ]`